src/pages/CreatePost.js

```diff
@@ -1,8 +1,23 @@
+import { useNavigate } from "react-router-dom";
+
 import PostForm from "../components/PostForm";
 
 function CreatePost() {
+  const navigate = useNavigate();
+
   function newPostHandler(postData) {
-      
+    fetch(
+      "https://friendface-react-default-rtdb.europe-west1.firebasedatabase.app/posts.json",
+      {
+        method: "POST",
+        body: JSON.stringify(postData),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }
+    ).then(() => {
+      navigate('/');
```

Comment:
I presume this makes the webpage go back to home? Does that mean you aren't implementing an SPA (Single Page Application)?

Reply:
From my understanding, navigate from the routing library is not really moving webpage, rather it is selecting a page/set of components to render - 'tricking' the user into thinking it's multiple pages https://www.split.io/blog/react-router-feature-flags/